testsuite/MDAnalysisTests/analysis/test_hbonds.py

```diff
@@ -47,58 +45,60 @@ def guess_types(names):
     return Atomtypes(np.array([guess_atom_type(name) for name in names], dtype=object))
 
 
-class TestHydrogenBondAnalysis(TestCase):
-    def setUp(self):
-        self.universe = u = MDAnalysis.Universe(PDB_helix)
-        self.kwargs = {
-            'selection1': 'protein',
-            'selection2': 'protein',
-            'detect_hydrogens': "distance",
-            'distance': 3.0,
-            'angle': 150.0,
-        }
-        # ideal helix with 1 proline:
-        self.values = {
-            'num_bb_hbonds':  u.atoms.n_residues - u.select_atoms('resname PRO').n_residues - 4,
-            'donor_resid': np.array([5,  6,  8,  9, 10, 11, 12, 13]),
+class TestHydrogenBondAnalysis(object):
+    @staticmethod
+    @pytest.fixture()
+    def universe():
+        return MDAnalysis.Universe(PDB_helix)
+
+    @staticmethod
+    @pytest.fixture()
+    def values(universe):
+        return {
+            'num_bb_hbonds': universe.atoms.n_residues - universe.select_atoms('resname PRO').n_residues - 4,
+            'donor_resid': np.array([5, 6, 8, 9, 10, 11, 12, 13]),
             'acceptor_resnm': np.array(['ALA', 'ALA', 'ALA', 'ALA', 'ALA', 'PRO', 'ALA', 'ALA'], dtype='U4'),
-            }
+        }
 
-    def _run(self, **kwargs):
+    kwargs = {
+        'selection1': 'protein',
+        'selection2': 'protein',
+        'detect_hydrogens': "distance",
+        'distance': 3.0,
+        'angle': 150.0,
+    }
+
+    def _run(self, universe, **kwargs):
```

Comment:
Shouldn't a method like this be replaced with a fixture?  This would then make it easier to cache this result (if possible) by increasing the scope